src/SwedbankPay.Sdk/CurrencyCode.cs

```diff
@@ -13,30 +13,43 @@
 using System.Globalization;
 using System.Linq;
 
-using Newtonsoft.Json;
-
 namespace SwedbankPay.Sdk
 {
+    /// <summary>
+    /// The currency code used for the payment
+    /// </summary>
     public class CurrencyCode
     {
-        [JsonConstructor]
         public CurrencyCode(string currencyCode)
         {
             if (string.IsNullOrEmpty(currencyCode))
+            {
                 throw new ArgumentNullException(nameof(currencyCode), "Currency code can't be null or empty");
+            }
+
             if (!IsValidCurrencyCode(currencyCode))
+            {
                 throw new ArgumentException($"Invalid currency code: {currencyCode}");
+            }
+
             Value = currencyCode;
         }
 
 
         private string Value { get; }
 
-
+        /// <summary>
+        /// Checks if the <paramref name="currencyCode"/> is valid.
+        /// This checks that it's of neutral culture, and use ISO currency symbols.
+        /// </summary>
+        /// <param name="currencyCode"></param>
+        /// <returns></returns>
         public static bool IsValidCurrencyCode(string currencyCode)
         {
             if (string.IsNullOrWhiteSpace(currencyCode))
+            {
                 return false;
+            }
 
             var regions = CultureInfo.GetCultures(CultureTypes.AllCultures)
```

Comment:
It seems unnecessary to recompute `regions` on every invocation, perhaps move it to a `static readonly` field that is initialized in a static constructor?